automation_tools/__init__.py

```diff
@@ -484,15 +484,39 @@ def setup_foreman_discovery():
 
 
 def setup_libvirt_key():
-    """Task to setup key pairs for secure comminication between Satellite server
-    and libvirt hypervisor."""
+    """Task to setup key pairs and verify host for secure communication between
+    Satellite server and libvirt hypervisor (qemu+ssh).
+
+    Expects the following environment variables:
+
+    LIBVIRT_HOSTNAME
+        hostname of libvirt hypervisor machine
+    LIBVIRT_KEY_URL
+        URL of relevant SSH key used for authentication to libvirt machine
+    """
+    libvirt_host = os.environ.get('LIBVIRT_HOSTNAME')
     key_url = os.environ.get('LIBVIRT_KEY_URL')
     key_file = '/root/.ssh/id_rsa'
+    foreman_key_file = '~foreman/.ssh/id_rsa'
     if key_url is None:
         print('You must specify the Libvirt key URL')
         sys.exit(1)
     run('wget -O {0} {1}'.format(key_file, key_url))
     run('chmod 600 {0}'.format(key_file))
+    sudo('wget -O {0} {1}'.format(foreman_key_file, key_url), user='foreman')
+    sudo('chmod 600 {0}'.format(foreman_key_file), user='foreman')
+    # Add libvirt host into known hosts
+    if libvirt_host is None:
```

Comment:
Move this check right after the libvirt_host is assigned. This to avoid any run/sudo operation to happen if not set.